lib/include/public/LogManagerBase.hpp

```diff
@@ -161,9 +177,16 @@ namespace MAT_NS_BEGIN
             return lock;
         }
 #endif
+        
+        static bool isGone()
+        {
+            return LogManagerExitHandler::isTerminated();            
+        }
 
         static inline bool isHost()
         {
+            if (isGone())
```

Comment:
Doesn't have to be thread-safe since the value of it is only changing when `atexit` processing is about to run. Won't ever flip to `false` during normal app lifecycle.